### PR TITLE
Remove ancient SNMM code in the MK3 branch

### PR DESCRIPTION
**Pull request by leptun**
Saved 258B of flash and 9B of ram from leftover code that wasn't behind ifdef. Finally the SNMM code is separated from the MMU2 code. The MK2 branch still has functional SNMM builds for the MK2.

### CODE REVIEW

**Comment by leptun** [comment]
@gudnimg If you want, can you please review the changes I did in this PR? There is mostly code removal, but also some refactoring of existing code.

File: Firmware/ultralcd.cpp [hunk 1]
```diff
@@ -5960,28 +5409,25 @@ uint8_t choose_menu_P(const char *header, const char *item, const char *last_ite
...
 char reset_menu() {
-    const uint8_t items_no =
-#ifdef SNMM
-        6;
-#else
-        5;
-#endif
-    static int8_t first = 0;
-    int8_t enc_dif = 0;
+	const uint8_t items_no = 5;
```
Comment (wavexx) [inline Firmware/ultralcd.cpp]: maybe we can move this after item[] and use ``sizeof(item)/sizeof(*item)``

**Review by wavexx — COMMENTED** [review]
Looking all good to me. A lot of cruft being removed.

File: Firmware/ultralcd.cpp (hunk 1 above)
Comment (leptun) [reply Firmware/ultralcd.cpp]: David said not to use *item in array size snippet and use item[0] instead. I'm updating that now

Comment (wavexx) [reply Firmware/ultralcd.cpp]: Should be fully equivalent semantically, right?
Either way is fine for me.

Comment (leptun) [reply Firmware/ultralcd.cpp]: Yes, it is equivalent.

Comment (leptun) [reply Firmware/ultralcd.cpp]: Pushed an update to this.

**Review by wavexx — APPROVED** [review]

**Comment by leptun** [comment]
Rebased and fixed merge conflicts from the usb timer PR

**Review by wavexx — APPROVED** [review]

**Review by gudnimg — CHANGES_REQUESTED** [review]
Can this be a for loop as you did for `MSG_LOAD_FILAMENT` above? 

```c
for (uint8_t i = 0; i < MMU_FILAMENT_COUNT; i++)
            MENU_ITEM_FUNCTION_NR_P(_T(MSG_EJECT_FILAMENT), i + '1', mmu_eject_filament, i); ////MSG_EJECT_FILAMENT c=16
```
Same comment for the `MSG_CUT_FILAMENT` menu items.

File: Firmware/ultralcd.cpp [hunk 2]
```diff
@@ -6101,11 +5536,11 @@ static void mmu_fil_eject_menu()
...
-        MENU_ITEM_FUNCTION_NR_P(_T(MSG_EJECT_FILAMENT), '1', mmu_eject_filament, 0);
-        MENU_ITEM_FUNCTION_NR_P(_T(MSG_EJECT_FILAMENT), '2', mmu_eject_filament, 1);
-        MENU_ITEM_FUNCTION_NR_P(_T(MSG_EJECT_FILAMENT), '3', mmu_eject_filament, 2);
-        MENU_ITEM_FUNCTION_NR_P(_T(MSG_EJECT_FILAMENT), '4', mmu_eject_filament, 3);
-        MENU_ITEM_FUNCTION_NR_P(_T(MSG_EJECT_FILAMENT), '5', mmu_eject_filament, 4);
+        MENU_ITEM_FUNCTION_NR_P(_T(MSG_EJECT_FILAMENT), '1', mmu_eject_filament, 0); ////MSG_EJECT_FILAMENT c=16
+        MENU_ITEM_FUNCTION_NR_P(_T(MSG_EJECT_FILAMENT), '2', mmu_eject_filament, 1); ////MSG_EJECT_FILAMENT c=16
+        MENU_ITEM_FUNCTION_NR_P(_T(MSG_EJECT_FILAMENT), '3', mmu_eject_filament, 2); ////MSG_EJECT_FILAMENT c=16
+        MENU_ITEM_FUNCTION_NR_P(_T(MSG_EJECT_FILAMENT), '4', mmu_eject_filament, 3); ////MSG_EJECT_FILAMENT c=16
+        MENU_ITEM_FUNCTION_NR_P(_T(MSG_EJECT_FILAMENT), '5', mmu_eject_filament, 4); ////MSG_EJECT_FILAMENT c=16
```
Comment (gudnimg) [inline Firmware/ultralcd.cpp]: See above comment. :)

**Comment by leptun** [comment]
Oh yeah, definitely. Forgot about those. Good thing you caught the issue in time.
Saved 134B extra just with those two menus 😲

**Comment by gudnimg** [comment]
Other than that the PR looks good to me 👍

**Review by wavexx — APPROVED** [review]

**Review by DRracer — APPROVED** [review]

**Pull request merged**